app/src/components/Chat/ChatContainer.jsx

```diff
@@ -6,24 +6,76 @@ import useArchive from "./../../hooks/useArchive";
 import autoScrollHistory from "./autoScrollHistory";
 import "./chat.css";
 
+const ChatInputStatus = ({ message }) => {
+  return <div className="send-status">{message}</div>;
+};
+
+const formatMessage = (eventObject) => {
+  // coerse String to Object
+  return typeof eventObject === "string" //
+    ? { text: `${eventObject || ""}`.trim() }
+    : eventObject;
+};
+
 const ChatContainer = ({ currentChannel, onChatExit }) => {
   const endOfChatLog = useRef(null);
+  const [archive, rewind] = useArchive(currentChannel);
+
   const [history, setHistory] = useState([]);
+  const [status, setStatus] = useState([]);
+  const [activity, setActivity] = useState();
 
-  useEffect(() => {
-    setHistory([]); // Reset history on channel change
-  }, [currentChannel]);
+  // Reset history on channel change
 
   const [channel] = useChannel(currentChannel, (message) => {
     setHistory((prev) => [...prev.slice(-199), message]);
   });
 
-  const [archive, rewind] = useArchive(currentChannel);
+  const sendMessage = (eventObject = null) => {
+    channel.publish("message", formatMessage(eventObject));
+    setStatusMessage(null);
+  };
 
-  const sendMessage = (messageText) => {
-    channel.publish("message", { text: messageText });
+  const sendStatus = (eventObject = null) => {
+    channel.presence.enter();
+    channel.presence.update(formatMessage(eventObject));
   };
 
+  const handlePresenceUpdate = (member) => {
+    const { data, clientId, connectionId } = member;
+    const { text } = data;
+
+    if (text === activity) return;
+
+    // clients on channel excluding the author as shallow copy Object
+    let clients = [];
+
+    channel.presence.get((_, members) => {
+      const typing = clientId === member.clientId;
```

Comment:
isn't this allows going to be true? clientId was defined on line 45 from member and you're then just comparing it back on itself again